### ISSUE TITLE
Condition: <list> on Resource triggers "Unknown exception while processing rule"

### ISSUE THREAD

**Issue by trav-c** [aws-cloudformation/cfn-python-lint#952]
*cfn-lint version: (`cfn-lint --version`)*
cfn-lint 0.21.4

*Description of issue.*
A resource with a Condition property defined as a list triggers:
```
E0002 Unknown exception while processing rule E8002: unhashable type: 'list_node'
/tmp/cfn-lint-condition-list-error.yaml:1:1
```
I believe that the use of lists / multiple values for a Condition property of a Resource is probably not legal (although I was unable to find clear confirmation of that in the documentation during a quick scan), but it should probably trigger a lint error rather than an exception.

It would also be helpful, if possible, to include the template line-number where the exception was triggered, rather than line:char 1:1 to make tracking the cause of such problems easier.

I have also seen the same exception, but for rule W1001, I though it was the same cause, but my reproduce test didn't re-trigger the W1001 case.

*Reproduce example*
```
AWSTemplateFormatVersion: 2010-09-09
Description: "cfn-lint condition list error"

Conditions:
    Cond1: !Equals [ !Ref 'AWS::Region', 'us-east-1' ]
    Cond2: !Equals [ !Ref 'AWS::Region', 'eu-west-1' ]

Resources:

    EIP1:
        Type: AWS::EC2::EIP
        Condition:
          - Cond1
          - Cond2
        Properties:
            Domain: 'vpc'

    EIP2:
        Type: AWS::EC2::EIP
        Condition: Cond1
        Properties:
            Domain: 'vpc'

    EIP3:
        Type: AWS::EC2::EIP
        Condition: Cond2
        Properties:
            Domain: 'vpc'
```
